### PR TITLE
Fixes csv bug and formatting / add test

### PR DESCRIPTION
**Pull request by JGSweets**
Fixes: https://github.com/capitalone/DataProfiler/issues/190

Header will still be auto detected even if user sets it to `auto`

### CODE REVIEW

File: dataprofiler/tests/data_readers/test_csv_data.py [hunk 1]
```diff
@@ -321,24 +321,41 @@ def _test_options(option, valid, invalid, expected_error):
...
+            invalid=["error", CSVData(), 1, None],
+            expected_error="'data_format' must be one of the following: ")
         
-        _test_options("selected_columns", valid = [['hello', 'world'], ["test"], []],
-                      invalid = ["error", CSVData(), 1, None],
-                      expected_error = "'selected_columns' must be a list") 
+        _test_options(
+            "selected_columns", valid=[['hello', 'world'], ["test"], []],
+            invalid=["error", CSVData(), 1, None],
+            expected_error="'selected_columns' must be a list")
```
Comment (JGSweets) [inline dataprofiler/tests/data_readers/test_csv_data.py:342]: formatting

File: dataprofiler/tests/data_readers/test_csv_data.py [hunk 2]
```diff
@@ -321,24 +321,41 @@ def _test_options(option, valid, invalid, expected_error):
...
         
-        _test_options("selected_columns", valid = [], invalid = [[0,1,2,3]],
-                      expected_error = "'selected_columns' must be a list of strings")
+        _test_options(
+            "selected_columns", valid=[], invalid=[[0,1,2,3]],
+            expected_error="'selected_columns' must be a list of strings")
```
Comment (JGSweets) [inline dataprofiler/tests/data_readers/test_csv_data.py:346]: formatting

File: dataprofiler/tests/data_readers/test_csv_data.py [hunk 3]
```diff
@@ -321,24 +321,41 @@ def _test_options(option, valid, invalid, expected_error):
...
+        filepath = "../data/csv/aws_honeypot_marx_geo.csv"
+        options = {'header': 'auto', 'delimiter': 'auto'}  # default value
+        data = CSVData(options=options)
+        self.assertEqual('auto', data.header)
+        self.assertFalse(data._checked_header)
+
+        data = CSVData(filepath, options=options)
+        retrieve_data = data.data
+        self.assertEqual(0, data.header)
+        self.assertTrue(data._checked_header)
```
Comment (JGSweets) [inline dataprofiler/tests/data_readers/test_csv_data.py:358]: new test

File: dataprofiler/data_readers/csv_data.py [hunk 4]
```diff
@@ -71,7 +71,7 @@ def __init__(self, input_file_path=None, data=None, options=None):
         self._quotechar = options.get("quotechar", None)
         self._selected_columns = options.get("selected_columns", list())
         self._header = options.get("header", 'auto')
-        self._checked_header = "header" in options
+        self._checked_header = "header" in options and self._header != 'auto'
```
Comment (JGSweets) [inline dataprofiler/data_readers/csv_data.py:74]: this is the only change, everything else is formatting

**Review by lettergram — APPROVED** [review]

**Review by lettergram — APPROVED** [review]

File: dataprofiler/tests/data_readers/test_csv_data.py [hunk 5]
```diff
@@ -321,24 +322,41 @@ def _test_options(option, valid, invalid, expected_error):
...
+            expected_error="'selected_columns' must be a list")
         
-        _test_options("selected_columns", valid = [], invalid = [[0,1,2,3]],
-                      expected_error = "'selected_columns' must be a list of strings")
+        _test_options(
+            "selected_columns", valid=[], invalid=[[0,1,2,3]],
+            expected_error="'selected_columns' must be a list of strings")
+
+        # test edge case for header being set
+        filepath = os.path.join(self.test_dir, 'csv/aws_honeypot_marx_geo.csv')
```
Comment (JGSweets) [inline dataprofiler/tests/data_readers/test_csv_data.py:350]: test here

**Review by gme5078 — DISMISSED** [review]

**Review by ChrisWallace2020 — APPROVED** [review]

**Review by gme5078 — APPROVED** [review]
